databricks/koalas/internal.py

```diff
@@ -381,6 +383,14 @@ def __init__(self, sdf: spark.DataFrame,
                                         len(set(len(i) for i in column_index)) <= 1)
         self._column_index = column_index
 
+        if isinstance(column_names, list):
+            if all(name is None for name in column_names):
+                self._column_names = None
+            else:
+                self._column_names = column_names
```

Comment:
Shall we explicitly check the expected types or the length of `column_names`?

Reply:
I used `is_list_like()` to check the type. Inside MultiIndex they check the if the length of column_index_names is equal to the length of the level. 
I think just checking the type is enough.